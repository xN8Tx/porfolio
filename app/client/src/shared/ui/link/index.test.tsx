import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import { NextIntlClientProvider } from "next-intl";
import { Link } from "./";

describe("UI/Link Testing", () => {
  test("should render with _blank target", () => {
    render(
      <NextIntlClientProvider messages={{}} locale="en">
        <Link href={"/link"} isExternal={true}>
          Children
        </Link>
      </NextIntlClientProvider>,
    );

    const link = screen.getByTestId<HTMLAnchorElement>("ui-link");
    expect(link.href).toBe("http://localhost/en/link");
    expect(link.textContent).toBe("Children");
    expect(link.target).toBe("_blank");
  });

  test("should render with _self target", () => {
    render(
      <NextIntlClientProvider messages={{}} locale="en">
        <Link href={"/link"} isExternal={false}>
          Children
        </Link>
      </NextIntlClientProvider>,
    );

    const link = screen.getByTestId<HTMLAnchorElement>("ui-link");
    expect(link.href).toBe("http://localhost/en/link");
    expect(link.textContent).toBe("Children");
    expect(link.target).toBe("_self");
  });

  test("should render with uppercase", () => {
    const tree = render(
      <NextIntlClientProvider messages={{}} locale="en">
        <Link href={"/link"} isExternal={false} uppercase={true}>
          Children
        </Link>
      </NextIntlClientProvider>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("should render with light", () => {
    const tree = render(
      <NextIntlClientProvider messages={{}} locale="en">
        <Link href={"/link"} isExternal={false} color="light">
          Children
        </Link>
      </NextIntlClientProvider>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("should render with background", () => {
    const tree = render(
      <NextIntlClientProvider messages={{}} locale="en">
        <Link href={"/link"} isExternal={false} color="background">
          Children
        </Link>
      </NextIntlClientProvider>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("should render with default", () => {
    const tree = render(
      <NextIntlClientProvider messages={{}} locale="en">
        <Link href={"/link"} isExternal={false} color="default">
          Children
        </Link>
      </NextIntlClientProvider>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("should render with small", () => {
    const tree = render(
      <NextIntlClientProvider messages={{}} locale="en">
        <Link href={"/link"} isExternal={false} size="small">
          Children
        </Link>
      </NextIntlClientProvider>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("should render with regular", () => {
    const tree = render(
      <NextIntlClientProvider messages={{}} locale="en">
        <Link href={"/link"} isExternal={false} size="regular">
          Children
        </Link>
      </NextIntlClientProvider>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("should render with large", () => {
    const tree = render(
      <NextIntlClientProvider messages={{}} locale="en">
        <Link href={"/link"} isExternal={false} size="large">
          Children
        </Link>
      </NextIntlClientProvider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
