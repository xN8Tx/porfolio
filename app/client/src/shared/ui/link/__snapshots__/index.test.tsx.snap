// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Link Testing should render with background 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="ui-link__link regular background"
        data-testid="ui-link"
        href="/en/link"
        target="_self"
      >
        Children
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="ui-link__link regular background"
      data-testid="ui-link"
      href="/en/link"
      target="_self"
    >
      Children
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI/Link Testing should render with default 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="ui-link__link regular default"
        data-testid="ui-link"
        href="/en/link"
        target="_self"
      >
        Children
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="ui-link__link regular default"
      data-testid="ui-link"
      href="/en/link"
      target="_self"
    >
      Children
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI/Link Testing should render with large 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="ui-link__link large default"
        data-testid="ui-link"
        href="/en/link"
        target="_self"
      >
        Children
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="ui-link__link large default"
      data-testid="ui-link"
      href="/en/link"
      target="_self"
    >
      Children
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI/Link Testing should render with light 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="ui-link__link regular light"
        data-testid="ui-link"
        href="/en/link"
        target="_self"
      >
        Children
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="ui-link__link regular light"
      data-testid="ui-link"
      href="/en/link"
      target="_self"
    >
      Children
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI/Link Testing should render with regular 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="ui-link__link regular default"
        data-testid="ui-link"
        href="/en/link"
        target="_self"
      >
        Children
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="ui-link__link regular default"
      data-testid="ui-link"
      href="/en/link"
      target="_self"
    >
      Children
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI/Link Testing should render with small 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="ui-link__link small default"
        data-testid="ui-link"
        href="/en/link"
        target="_self"
      >
        Children
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="ui-link__link small default"
      data-testid="ui-link"
      href="/en/link"
      target="_self"
    >
      Children
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI/Link Testing should render with uppercase 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="ui-link__link regular default uppercase"
        data-testid="ui-link"
        href="/en/link"
        target="_self"
      >
        Children
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="ui-link__link regular default uppercase"
      data-testid="ui-link"
      href="/en/link"
      target="_self"
    >
      Children
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
